lib/model/message_list.dart

```diff
@@ -155,7 +155,7 @@ mixin _MessageSequence {
         }
       case MessageListRecipientHeaderItem(:var message):
       case MessageListDateSeparatorItem(:var message):
-        return (message.id <= messageId) ? -1 : 1;
+        return message.id != null && message.id! <= messageId ? -1 : 1;
```

Comment:
nit: I think this condition would be clearer by splitting up onto two lines:
```suggestion
        if (message.id == null)                  return 1;
        return message.id! <= messageId ? -1 : 1;
```

(then the spacing is to match the alignment of other cases in this method)